Sources/Datadog/FeaturesIntegration/CITestIntegration.swift

```diff
@@ -0,0 +1,81 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+private enum DDCFMessageID {
+    static let customTags: Int32 = 0x1111
+    static let enableRUM: Int32 = 0x2222
+    static let forceFlush: Int32 = 0x3333
+}
+
+internal enum CITestIntegration {
+    /// CI context to be attached to RUM events that identifies it was started by a CI test,
+    static var ciTestExecutionID: String? {
+        return ProcessInfo.processInfo.environment["CI_VISIBILITY_TEST_EXECUTION_ID"]
+    }
+
+    /// Convinence function to check if RUM was started in a CIApp test
+    static func isEnabled() -> Bool {
+        return ciTestExecutionID != nil
+    }
+
+    /// Origin value that must be set to Spans and headers to indicate that the trace was initiated by a CIApp test
+    static var origin: String? {
+        if isEnabled() {
+            return "ciapp-test"
+        }
+        return nil
+    }
```

Comment:
This static APIs model is slightly inconvenient and can be easily overlooked. Now it's a bit unbalanced:
* `isEnabled()` is considered in callee when computing `origin`,
* `isEnabled()` is considered in caller when calling `startIntegration()` in `RUMMonitor`.

How about changing it to object-oriented way, where "integration is not enabled" is indicated by `nil` value, e.g.:
```swift
internal class CITestIntegration {
   /// Current and active integration with CIApp. 
   /// `nil` if the integration is not enabled.
   static let active: CITestIntegration? = CITestIntegration()

   let ciTestExecutionID: String
   let origin = "ciapp-test"

   private init?(processInfo: ProcessInfo = .processInfo) {
      guard let testID = processInfo.environment["CI_VISIBILITY_TEST_EXECUTION_ID"] else {
         return nil
      }
      self.ciTestExecutionID = testID
   }

   func startIntegration() { /* ... */ }
}
```

This way the caller is given streamlined API and cannot make mistake on calling sth when this integration is disabled:
```swift
CITestIntegration.active?.startIntegration()
let origin = CITestIntegration.active?.origin
let testID = CITestIntegration.active?.ciTestExecutionID
// ...
```

Reply:
Good idea, looks much better. Will change to it

Reply:
I have updated it with the last commit, code is much cleaner now